ibllib/qc/task_extractors.py

```diff
@@ -64,31 +64,42 @@ def _ensure_required_data(self):
         :return:
         """
         dstypes = [
-            "_iblrig_taskData.raw.jsonable",
-            "_iblrig_taskSettings.raw.json",
-            "_iblrig_encoderPositions.raw.ssv",
-            "_iblrig_encoderEvents.raw.ssv",
-            "_iblrig_stimPositionScreen.raw.csv",
-            "_iblrig_syncSquareUpdate.raw.csv",
-            "_iblrig_encoderTrialInfo.raw.ssv",
-            "_iblrig_ambientSensorData.raw.jsonable",
+            "_iblrig_taskData.raw",
+            "_iblrig_taskSettings.raw",
+            "_iblrig_encoderPositions.raw",
+            "_iblrig_encoderEvents.raw",
+            "_iblrig_stimPositionScreen.raw",
+            "_iblrig_syncSquareUpdate.raw",
+            "_iblrig_encoderTrialInfo.raw",
+            "_iblrig_ambientSensorData.raw",
         ]
         eid = self.one.path2eid(self.session_path)
         self.log.info(f"Downloading data for session {eid}")
-        collections = ['raw_behavior_data'] * len(dstypes)
+        collections = ('raw_behavior_data',)
         # Ensure we have the settings
         settings, _ = self.one.load_datasets(eid, ["_iblrig_taskSettings.raw.json"],
                                              collections=['raw_behavior_data'],
                                              download_only=True, assert_present=False)
         if settings and get_session_extractor_type(self.session_path) == 'ephys':
-            dstypes.extend(['_spikeglx_sync.channels.npy',
-                            '_spikeglx_sync.polarities.npy',
-                            '_spikeglx_sync.times.npy',
-                            '_spikeglx_ephysData_g0_t0.nidq.meta',
-                            '_spikeglx_ephysData_g0_t0.nidq.wiring.json'])
-            collections = collections + ['raw_ephys_data'] * 5
-        files, _ = self.one.load_datasets(eid, dstypes, collections=collections,
-                                          download_only=True, assert_present=False)
+
+            dstypes.extend(['_spikeglx_sync.channels',
+                            '_spikeglx_sync.polarities',
+                            '_spikeglx_sync.times',
+                            'ephysData.raw.meta',
+                            'ephysData.raw.wiring'])
+
+            if len(self.one.list_datasets(eid, collection='raw_ephys_data')) == 0:
+                collections += ('raw_ephys_data/probe00', 'raw_ephys_data/probe01')
+            else:
+                collections += ('raw_ephys_data',)
+
+        files = []
+        for dstype in dstypes:
+            dataset = self.one.datasets_from_type(eid, dstype, full=True)
```

Comment:
Tomorrow this will be `dataset = self.one.type2datasets(eid, dstype, details=True)`.  You could remove the for loop as now it permits a list of dataset types (should be quicker)